packages/synapse-react-client/src/components/OAuthClientManagement/CreateOAuthClient.integration.test.tsx

```diff
@@ -35,9 +43,10 @@ const defaultProps: CreateOAuthModalProps = {
   setIsShowingConfirmModal: jest.fn(),
 }
 
-jest
-  .spyOn(SynapseClient, 'isOAuthClientReverificationRequired')
-  .mockReturnValue(Promise.resolve({ reverificationRequired: true }))
```

Comment:
I think tests were flaky because the call to `isOAuthClientReverificationRequired` is wrapped in `useDebouncedEffect`, so depending on the test timing, the service might be called and cause the warning modal to appear when reverification was not expected (Creates a client on save, Updates a client) or not be called when the warning modal was expected to appear (show warning modal when changing redirect uri).